### ISSUE TITLE
Fantomas scrambled up a huge script file

### ISSUE THREAD

**Issue by halcwb** [fsprojects/fantomas-for-vs#34]
Unfortunately fantomas set up to run on save, scrambled a huge script [file](https://gist.github.com/halcwb/eaad0977d962f17aaee78c18e6670a48) of mine. This has caused literally days of work loss. 

Is there a way to recover from this?

Well, actually, I am quite distressed by all of this. This is causing me big problems!

**Comment by dsyme** [comment]
Does your OS or editor keep intermediate copies or history? Undo in the editor?

**Comment by halcwb** [comment]
Unfortunately, I set fantomas to format on save. And normally I do a lot of intermediate commits. Only, as bad luck has it, working on a recoding of my project putting everything in the huge script file, at almost the finish, the problem occurred. So, I am afraid, I have to do it all over.

**Comment by halcwb** [comment]
The least fantomas should do is check whether the formatted code is still syntactically correct. If it isn't syntactically correct do not format the code but leave it as it is. This could prevent the mess that happened to me.

**Comment by baronfel** [comment]
A couple questions that need to be answered:
* does standalone fantomas (i.e. the fantomas dotnet tool) have the same issue
* if so, then yes this is a fantomas issue and needs to be solved there
* if not, then is the issue in this repo, inside the code that gets the fantomas response and issues edits to the editor? or is it inside the editor itself after the edit has been issued by this extension?

**Comment by halcwb** [comment]
I tried to undo, but that didn't result in a recovery. I did this before giving up and closing the file.

**Comment by deviousasti** [comment]
@nojaf Do we check after formatting if the CodeFormatter produced a valid output?

**Comment by nojaf** [comment]
I don't think so for `Fantomas.Client`, we do this for the command line tool though.
